src/configupdater/parser.py

```diff
@@ -365,10 +465,12 @@ def _read(self, fp: Iterable[str], fpname: str):
                         # optname = self.optionxform(optname.rstrip())
                         # keep original case of key
                         optname = optname.rstrip()
+                        if sectname is None:
+                            msg = f"Could not find the section name for {optname}"
+                            raise InconsistentStateError(msg, fpname, lineno, line)
```

Comment:
This is a check required by the type checker. According to the type analysis, `sectname` could be `None` here (and the state machine behind the parser is to complex for me to figure out, so I just added runtime checks).